src/EnergyPlus/SolarShading.cc

```diff
@@ -5818,7 +5818,10 @@ namespace SolarShading {
 
 						//Also update the nominal diffuse transmittance
 						NomDiffTrans = SurfaceWindow( SurfNum ).ComplexFen.State( SurfaceWindow( SurfNum ).ComplexFen.CurrentState ).WinDiffTrans;
-						Construct( Surface( SurfNum ).Construction ).TransDiff = NomDiffTrans;
+
+						// Do not store in TransDiff because it is not used by BSDF and rest of the code uses it as flag for opaque
+						// surface incorrectly assuming wall heat transfer routines for windows.
+						// Construct( Surface( SurfNum ).Construction ).TransDiff = NomDiffTrans;
```

Comment:
Yes, we have this all over the code:
if ( Construct( ConstrNum ).TransDiff <= 0.0 ) { // Opaque surface

Unfortunately, for window with shading devices this can be true statement. I decided not to fix this bug for other window models yet, but we should definitely put it into some bug list.

Reply:
Issue #6315 posted for the Constuct.TransDiff triggering opaque wall heat transfer problem.